src/main/resources/META-INF/resources/primefaces/tristatecheckbox/tristatecheckbox.js

```diff
@@ -91,8 +91,10 @@ PrimeFaces.widget.TriStateCheckbox = PrimeFaces.widget.BaseWidget.extend({
 
             // change title to the new one
             var iconTitles = this.box.data('titlestates');
-            if(iconTitles!=null && iconTitles.length>0){
-                this.box.attr('title', iconTitles[newValue]);
+            if(iconTitles!=null) {
+                if(iconTitles.titles!=null && iconTitles.titles.length>0) {
```

Comment:
Why is this iconTitles != null done twice in a row?

Reply:
Sorry? The second one tests for iconTitles.**titles**. This has been introduced due to the problem mentioned above.

We could however compact the code into a single line...

Reply:
No I think a single line makes it messier to read. I am OK with 2 lines just my eyes didn't see the ".title" in the 2nd line.

Reply:
Sorry, too late. Just committed...